### PR TITLE
Deduplicate MAU events using CloudAnonymizationKey

### PR DESCRIPTION
**Pull request by michellescripts**
Cloud now surfaces a CloudAnonymizationKey to Teleport via the features endpoint. This key, when present, should be used to anonymize usage events.

The features service checks for updates to this key on a regular interval, and if changed, passes it to the anonymizer which is used in both the usage reporter and the aggregate usage reporter.

When not present, we fall back to the pre-existing logic to use either the 1. license anonymization key or the 2. cluster ID.

e-sibling: https://github.com/gravitational/teleport.e/pull/5051
Supports: https://github.com/gravitational/cloud/issues/9458

### CODE REVIEW

File: lib/auth/auth.go [hunk 1]
```diff
@@ -1894,6 +1894,12 @@ func (a *Server) GetAnonymizationKey(ctx context.Context, opts ...services.Marsh
 		return a.GetClusterID(ctx, opts...)
 	}
 
+	// check for cloud anonymization key
+	key := modules.GetModules().Features().CloudAnonymizationKey
+	if len(key) != 0 {
+		return string(key), nil
+	}
+
 	return string(a.license.AnonymizationKey), nil
```
Comment (zmb3) [inline lib/auth/auth.go:1903]:
```suggestion
	return cmp.Or(
		modules.GetModules().Features().CloudAnonymizationKey,
		string(a.license.AnonymizationKey))
```

File: api/proto/teleport/legacy/client/proto/authservice.proto [hunk 2]
```diff
@@ -581,6 +581,8 @@ message Features {
   // NOTE: this flag is used to signal that Access Monitoring is *enabled* on a cluster.
   // *Access* to the feature is gated on the `AccessMonitoring` entitlement.
   bool AccessMonitoringConfigured = 36;
+  // cloud_anonymization_key is a hash of the Salesforce ID used to anonymize usage events
+  bytes cloud_anonymization_key = 37 [json_name = "cloudAnonymizationKey"];
```
Comment (zmb3) [inline api/proto/teleport/legacy/client/proto/authservice.proto:585]:
```suggestion
  bytes CloudAnonymizationKey = 37 [ (gogoproto.jsontag) = "cloud_anonymization_key"];
```

Can we be consistent with the other fields in this message? (assuming we can change the JSON key without breaking anything)

File: lib/modules/modules.go [hunk 3]
```diff
@@ -59,6 +59,9 @@ type Features struct {
 	SupportType proto.SupportType
 	// Entitlements reflect Cloud Entitlements including access and limits
 	Entitlements map[entitlements.EntitlementKind]EntitlementInfo
+	// CloudAnonymizationKey is the key used to anonymize usage events in a cluster. If not present, defaults to
+	// cluster id.Only applicable for Cloud hosted customers, not self-hosted.
```
Comment (zmb3) [inline lib/modules/modules.go:63]:
```suggestion
	// cluster ID. Only applicable for Cloud customers (self-hosted clustsers
	// get their anonymization key from the license file).
```

File: lib/auth/auth.go [hunk 4]
```diff
@@ -2004,11 +2004,15 @@ func (a *Server) GetClusterID(ctx context.Context, opts ...services.MarshalOptio
 // GetAnonymizationKey returns the anonymization key that identifies this client.
 // It falls back to the cluster ID if the anonymization key is not set in license file.
```
Comment (zmb3) [inline lib/auth/auth.go:2005]: Should we update this comment? Maybe:
```suggestion
// The anonymization key may be any of the following, in order of precedence:
// - (Teleport Cloud) a key provided by the Teleport Cloud API
// - a key embedded in the license file
// - the cluster's UUID
```

File: lib/auth/auth_test.go [hunk 5]
```diff
@@ -4334,6 +4334,65 @@ func TestCleanupNotifications(t *testing.T) {
...
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testAuthServer, err := NewTestAuthServer(TestAuthServerConfig{
+				Dir:       t.TempDir(),
+				Clock:     clockwork.NewFakeClock(),
+				ClusterID: "cluster-id",
+			})
+			require.NoError(t, err)
+			testTLSServer, err := testAuthServer.NewTestTLSServer()
```
Comment (zmb3) [inline lib/auth/auth_test.go:4382]:
```suggestion
			t.Cleanup(func() { require.NoError(t, testAuthServer.Close()) })
			testTLSServer, err := testAuthServer.NewTestTLSServer()
```

File: lib/auth/auth_test.go [hunk 6]
```diff
@@ -4334,6 +4334,65 @@ func TestCleanupNotifications(t *testing.T) {
...
+			require.NoError(t, err)
```
Comment (zmb3) [inline lib/auth/auth_test.go:4383]:
```suggestion
			require.NoError(t, err)
			t.Cleanup(func() { require.NoError(t, tlsServer.Close()) })
```

File: lib/modules/modules.go [hunk 7]
```diff
@@ -58,6 +58,10 @@ type Features struct {
 	SupportType proto.SupportType
 	// Entitlements reflect Cloud Entitlements including access and limits
 	Entitlements map[entitlements.EntitlementKind]EntitlementInfo
+	// CloudAnonymizationKey is the key used to anonymize usage events in a cluster. If not present, defaults to
+	// cluster ID. Only applicable for Cloud customers (self-hosted clusters get their anonymization key from the
+	// license file).
+	CloudAnonymizationKey []byte
```
Comment (zmb3) [inline lib/modules/modules.go:64]: I would simplify the comments here if you accept my previous suggestion to expand the comments on the `GetAnonymizationKey` method.
```suggestion
	// CloudAnonymizationKey is the key used to anonymize usage events in a cluster.
	// Only applicable for Cloud customers (self-hosted clusters get their anonymization key from the
	// license file).
	CloudAnonymizationKey []byte
```

**Review by zmb3 — APPROVED** [review]
LGTM, just a couple small suggestions.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**